sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/NonSessionMessagePump.java

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.implementation.instrumentation.ReceiverKind;
+import com.azure.messaging.servicebus.implementation.instrumentation.ServiceBusReceiverInstrumentation;
+import org.reactivestreams.Publisher;
+import reactor.core.Disposable;
+import reactor.core.Disposables;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import static com.azure.core.amqp.implementation.ClientConstants.ENTITY_PATH_KEY;
+import static com.azure.core.amqp.implementation.ClientConstants.FULLY_QUALIFIED_NAMESPACE_KEY;
+import static com.azure.core.amqp.implementation.ClientConstants.PUMP_ID_KEY;
+
+/**
+ * Abstraction to pump messages using a {@link ServiceBusReceiverAsyncClient} (associated with a session unaware entity)
+ * as long as the client is healthy.
+ *
+ * <p>The pumping starts upon subscribing to the Mono that {@link NonSessionMessagePump#begin} returns. The pumping can
+ * be stopped by cancelling the subscription that {@link NonSessionMessagePump#begin} returned.</p>
+ *
+ * <p>Once the Mono from {@link NonSessionMessagePump#begin} terminates, to restart the pumping a new {@link NonSessionMessagePump}
+ * instance should be obtained.</p>
+ *
+ * <p>The abstraction {@link ServiceBusProcessor} takes care of managing a {@link NonSessionMessagePump} and obtaining
+ * the next NonSessionMessagePump when current one terminates.</p>
+ */
+final class NonSessionMessagePump {
+    private static final AtomicLong COUNTER = new AtomicLong();
+    private static final Duration CONNECTION_STATE_POLL_INTERVAL = Duration.ofSeconds(20);
+    private final long  pumpId;
+    private final ServiceBusReceiverAsyncClient client;
+    private final String fqdn;
```

Comment:
same nit with fqdn.  But I am usually not a fan of abbreviations when they can be typed.. and with auto-complete. 😅

Reply:
**[DONE]**